src/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -56,7 +56,7 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource *source, bool
   else
   {
     //QgsDebugMsg( "Feature iterator of " + mSource->mLayerName + ": acquiring connection");
-    mConn = QgsOgrConnPool::instance()->acquireConnection( QgsOgrProviderUtils::connectionPoolId( mSource->mDataSource ) );
+    mConn = QgsOgrConnPool::instance()->acquireConnection( QgsOgrProviderUtils::connectionPoolId( mSource->mDataSource ), request.connectionTimeout(), request.requestMayBeNested() );
```

Comment:
Better test this - I tried (very briefly) implementing the timeout for ogr a month or so back, and found that the invalid iterator handling caused a crash

Reply:
Any test in particular to run?

Reply:
I honestly can't remember - but it was as soon as the timeout was experienced I'd get a crash. Try with a tiny timeout and a remote layer and you should hit it.

Reply:
Thanks, should be fixed?